### ISSUE TITLE
Completion not working as expected

### ISSUE THREAD

**Issue by tom-james-watson** [neoclide/coc.nvim#146]
Here's my barebones vim config I've used to test this:

```
call plug#begin('~/.vim/plugged')

Plug 'neoclide/coc.nvim', {'do': { -> coc#util#install()}}

call plug#end()

" Use tab for trigger completion with characters ahead and navigate.
" Use command ':verbose imap <tab>' to make sure tab is not mapped by other plugin.
inoremap <silent><expr> <TAB>
      \ pumvisible() ? "\<C-n>" :
      \ <SID>check_back_space() ? "\<TAB>" :
      \ coc#refresh()
inoremap <expr><S-TAB> pumvisible() ? "\<C-p>" : "\<C-h>"

function! s:check_back_space() abort
  let col = col('.') - 1
  return !col || getline('.')[col - 1]  =~# '\s'
endfunction

" Use <c-space> for trigger completion.
inoremap <silent><expr> <c-space> coc#refresh()

" Use <cr> for confirm completion.
" Coc only does snippet and additional edit on confirm.
inoremap <expr> <cr> pumvisible() ? "\<C-y>" : "\<C-g>u\<CR>"
```
I then install the following:

```
:CocInstall coc-eslint coc-tsserver
```

I then am not getting any kind of useful completion:

![image](https://user-images.githubusercontent.com/3852719/48076065-c34a0000-e1e4-11e8-830d-94c4f89982b5.png)

node - v10.8.0
nvim - v0.3.1

Using `coc-pyls` with Python works exactly as I'd expect, with proper semantic completion suggestions.

Am I missing something?

**Comment by chemzqm** [comment]
There's a bug with diff content on 0.3.0, you can add an empty line to avoid this issue.
A new release would be published later.

**Comment by tom-james-watson** [comment]
Mm this happens regardless of whether the current line is the last in the file:

![screenshot 2018-11-07 at 00 05 43](https://user-images.githubusercontent.com/3852719/48099763-eb575480-e220-11e8-9b52-d52be3d51a81.png)

**Comment by chemzqm** [comment]
@tom-james-watson I can confirm that, build from source code to avoid this issue.

**Comment by tom-james-watson** [comment]
OK - can confirm that fixes it. Thanks!

**Comment by chemzqm** [comment]
A new release is available.

**Comment by chemzqm** [comment]
Fix on 0.0.31
